com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -724,7 +770,62 @@ void ExecuteRenderPass(ScriptableRenderContext context, ScriptableRenderPass ren
             context.ExecuteCommandBuffer(cmd);
             CommandBufferPool.Release(cmd);
 
-            renderPass.Execute(context, ref renderingData);
+            if (IsRenderPassEnabled(renderPass) && cameraData.cameraType == CameraType.Game)
+            {
+                int validColorBuffersCount = (int)RenderingUtils.GetValidColorBufferCount(renderPass.colorAttachments);
+
+                bool isLastPass = renderPass.srpDescriptor.isLastPass;
+                bool isLastPassToBB = isLastPass && (m_ActiveColorAttachmentDescriptors[0].loadStoreTarget == BuiltinRenderTextureType.CameraTarget);
+                bool useDepth = m_ActiveDepthAttachment == RenderTargetHandle.CameraTarget.Identifier() && (!(isLastPassToBB || (isLastPass && cameraData.camera.targetTexture != null)));
+
+                var attachments =
+                    new NativeArray<AttachmentDescriptor>(useDepth && !renderPass.srpDescriptor.isDepthOnly ? validColorBuffersCount + 1 : 1, Allocator.Temp);
+
+                for (int i = 0; i < validColorBuffersCount; ++i)
+                    attachments[i] = m_ActiveColorAttachmentDescriptors[i];
+
+                if (useDepth && !renderPass.srpDescriptor.isDepthOnly)
+                    attachments[validColorBuffersCount] = m_ActiveDepthAttachmentDescriptor;
+
+                var desc = renderingData.cameraData.cameraTargetDescriptor;
+                var sampleCount = desc.msaaSamples;
+                int width = renderPass.srpDescriptor.width != -1 ? renderPass.srpDescriptor.width : desc.width;
+                int height = renderPass.srpDescriptor.height != -1 ? renderPass.srpDescriptor.height : desc.height;
+                sampleCount = renderPass.srpDescriptor.sampleCount != -1
+                    ? renderPass.srpDescriptor.sampleCount
+                    :
+#if UNITY_EDITOR
+                    !isLastPassToBB ? sampleCount : 1;
+#else
+                    //In case MSAA is disabled on camera, we need to still check the QualitySettings, as backbuffer will be created with that setting in mind
+                    isLastPassToBB? QualitySettings.antiAliasing : sampleCount;
+#endif
```

Comment:
This sampleCount logic and last pass is very convoluted.

Reply:
most of this stuff wasn't needed anymore and was related to an old FinalBlitPass conversion. Simplified the code